.github/workflows/test.yaml

```diff
@@ -20,22 +17,26 @@ jobs:
         with:
           submodules: "true"
 
-      - uses: actions/setup-python@v5
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
         with:
-          python-version: "3.10"
+          uv-version: "0.4.22"
 
-      - name: Install
-        run: pip install -r requirements/test.txt
+      - name: Set up Python
+        uses: actions/setup-python@v5
```

Comment:
We could probably replace this action with `uv python install` as a step, and it should have the same effect :)

Reply:
Actually I'm curious your thoughts on this. `uv python install` installs the most recent python version compatible with your rules in `pyproject.toml`. Where as using the setup option you can actually lock the version. Although we don't currently utilise that feature.

Do we have a preference for `uv python install`? I don't mind either way.

Reply:
So we can pin to a specific version with `uv python pin 3.10` and it will pin it to that within either the `uv.lock` or `pyproject.toml` I can't remember which. I think that should lock it in

Reply:
I think you would then be able to reduce the number of files we need to modify anytime we change versions. For e.g in the Makefile we can then change `uv venv --python=python3.10` to just `uv venv` since it should pick up the version from the toml or lock file

Reply:
Nice. I'll make that change. I'll remove the .python-version from the .gitignore then.

Reply:
You might not need the .python-version altogether I think 👀

Reply:
I ran the `uv python pin 3.10` and it generated the file. That must be the mechanism for locking the version. If it's not I'm happy to remove the file but I haven't looked into whether the file is unnecessary or not.

Reply:
Ah okay, my bad that must be it then!